pkg/ast/pipesearch/searchHandler.go

```diff
@@ -527,41 +530,64 @@ func convertQueryCountToTotalResponse(qc *structs.QueryCount) interface{} {
 }
 
 /*
+Function designed to parse Time from a String input.
    Supports "now-[Num][Unit]"
    Num ==> any positive integer
    Unit ==> m(minutes), h(hours), d(days)
 */
 
-func parseAlphaNumTime(nowTs uint64, inp string, defValue uint64) uint64 {
+func parseAlphaNumTime(nowTs uint64, inp string, defValue uint64) (uint64, error) {
 
+	//removing spaces if any from input
 	sanTime := strings.ReplaceAll(inp, " ", "")
 
+	//to be returned as is if no filters are applied
 	if sanTime == "now" {
-		return nowTs
+		return nowTs, nil
 	}
 
+	//setting return value as default
 	retVal := defValue
 
+	//regex pattern for "now-[number]" and unit at the end which only accepts 'm', 'h' and 'd'
+	pattern := `^now-\d+[mhd]$`
+	r, patternError := regexp.Compile(pattern)
+	if patternError != nil {
+		log.Errorf("startEpoch/endEpoch Input received is invalid. Value probably modified")
+		return retVal, fmt.Errorf("startEpoch Input received is invalid.")
+	}
```

Comment:
If you get an error here, it's because the `pattern` you defined is invalid regex, not because `inp` didn't match it. Since this `pattern` is hard coded and not provided by the user, you can use `regexp.MustCompile(pattern)` which will panic if there's an error instead of just return the error (this is only ok because we know exactly what the `pattern` is and it's not provided by the user, so we trust it).

Reply:
Yes, I was thinking about the best way to go about the string validation and regex made sense. I shall implement the MustCompile function

Reply:
I have replaced it with MustCompile function. I am getting an error with the test cases for this function, I did look into other MustCompile implementations in the codebase but I'm not sure exactly how to go about it. Could use some insights here @AndrewHess

Reply:
Your use of `regexp.MustCompile` looks correct to me. For the test that's failing, you can check the stack trace and it's an issue with line 86 in the test file. That line is:
```
assert.NotNil(t, nil, err, "expected=%v, actual=%v", expected, err)
```
I think you're just passing `nil` when you shouldn't. You don't even need the extra parameters like `expected=...` because if that test fails you should see that printed anyways. So you should be able to change that line to simply
```
assert.NotNil(t, err)
```

PS. You can run the whole unit test suite locally with `make test`

Reply:
Got it. I've replaced it to:
`assert.NotNil(t, err)`

On another note, while running `make test` , I am looking at some other errors (I just fast forward updated my repo). It looks like so:
`# go.opentelemetry.io/otel/internal/global
../../../go/pkg/mod/go.opentelemetry.io/otel@v1.24.0/internal/global/handler.go:44:18: undefined: atomic.Pointer
../../../go/pkg/mod/go.opentelemetry.io/otel@v1.24.0/internal/global/internal_logging.go:30:25: undefined: atomic.Pointer
note: module requires Go 1.20
# `

Is there some other issue going on?